sycl/doc/extensions/experimental/sycl_ext_oneapi_invoke_simd.asciidoc

```diff
@@ -123,10 +123,15 @@ of a `simd<T, N>` type (e.g. using multiple registers rather than an array).
 We expect implementations of this ABI to build on Clang's `ext_vector_type`,
 rather than specifying the register types directly.
 
-The value of `N` for each `simd` argument to a SIMD function must be the same,
-and represents the sub-group size of the calling kernel.  If a function accepts
-`simd` arguments with multiple `N` values, it cannot be invoked from a SPMD
-kernel and the implementation must issue a diagnostic in this case.
+Each of the following must hold true for a SIMD function and its invoke_simd
+call site:
+- the value of `N` of each `simd` formal parameter corresponding to a non-uniform
+SPMD actual argument (if any) must be the same
+- if the return type of the SIMD function is `simd<T, M>` or `simd_mask<T, M>`,
+  then `M` must match `N` above
+Otherwise, the SIMD function cannot be invoked from an SPMD kernel and the
+implementation must issue a diagnostic in this case.
```

Comment:
I think this is now inconsistent with the description on lines 283-294.  The text here implies that it's illegal to invoke a fully uniform function, and I prefer the description that comes later.  Do we need to duplicate it?

Reply:
I thought the two 'if's in the above bullets put requirements under two separate conditions, and if no conditions hold, then requirements are not imposed, and fully uniform can be called.

The lines 132-133 just try to tell what happens if the "must" requirement above does not hold, so I think this should remain in some form - ? I'll try to reformulate.